djangofiles/frisbeer/views.py

```diff
@@ -78,6 +78,11 @@ def create_teams(self, request, pk=None):
         print("Created")
         return redirect(reverse("frisbeer:games-detail", args=[pk]))
 
+    def destroy(self, request, *args, **kwargs):
+        if request.user and not request.user.is_superuser and self.get_object().players.count() > 0:
```

Comment:
This would be permitted if `request.user` is empty, which, I guess, is not desired. Anyway, does the `request.user` field always exist? Would raise `AttributeError` if not

Reply:
Django (rest) is responsible for handling authentication and authorization normally, this is just a special added case in addition to normal checks

Reply:
Without api token response is
 {
    "detail": "Authentication credentials were not provided."
}